provision.sh

```diff
@@ -159,30 +177,46 @@ rm spiff_linux_amd64.zip
 
 
 # Upload the bosh release, set the deployment, and execute
-bosh upload release https://bosh.io/d/github.com/cloudfoundry/cf-release?v=${cfReleaseVersion}
-bosh deployment cf-openstack-${CF_SIZE}
-bosh prepare deployment
+deployedVersion=$(bosh releases | grep " ${cfReleaseVersion}" | awk '{print $4}')
+deployedVersion="${deployedVersion//[^[:alnum:]]/}"
+if [[ ! "$deployedVersion" == "${cfReleaseVersion}" ]]; then
+  bosh upload release https://bosh.io/d/github.com/cloudfoundry/cf-release?v=${cfReleaseVersion}
+  bosh deployment cf-openstack-${CF_SIZE}
+  bosh prepare deployment || bosh prepare deployment  #Seems to always fail on the first run...
```

Comment:
I'm referring to `#Seems to always fail on the first run...`, also bosh prepare deployment is idempotent.